src/libraries/System.Text.Json/src/System/Text/Json/JsonConstants.cs

```diff
@@ -61,7 +61,12 @@ internal static partial class JsonConstants
         public const int MaxExpansionFactorWhileTranscoding = 3;
 
         // When transcoding from UTF8 -> UTF16, the byte count threshold where we rent from the array pool before performing a normal alloc.
-        public const long ArrayPoolMaxSizeBeforeUsingNormalAlloc = 1024 * 1024;
+        public const long ArrayPoolMaxSizeBeforeUsingNormalAlloc =
+#if NET6_0_OR_GREATER
+            1024 * 1024 * 1024;
```

Comment:
Could you add a comment explaining why a different constant is being used for builds >= 6.0?

(FWIW at this point this would be equivalent to using `#if NETCOREAPP`).

Reply:
```suggestion
            1024 * 1024 * 1024; // ArrayPool limit increased in .NET 6
```